File: server/src/main/java/gmetrica/enumeration/TipoContagem.java
package gmetrica.enumeration;

public enum TipoContagem {
	
	CONTAGEM_REFERENCIA ("Contagem de Referência"),
	ESTIMATIVA_INICIAL ("Estimativa Inicial"),
	CONTAGEM_ENCERRAMENTO ("Contagem de Encerramento");
	
	private String descricao;
	
	private TipoContagem(String descricao) {
		this.descricao = descricao;
	}

	public String getDescricao() {
		return descricao;
	}

	public static TipoContagem fromDescricao(String descricao) {
		if(descricao == null) {
			return null;
		}
		
		for(TipoContagem valor: TipoContagem.values()) {
			
			if(valor.getDescricao().equals(descricao)) {
				return valor;
			}
		}
		
		return null;
	}
	
	public static TipoContagem fromContains(String descricao) {
		if(descricao == null) {
			return null;
		}
		
		for(TipoContagem valor: TipoContagem.values()) {
			descricao = descricao.substring(0,1).toUpperCase().concat(descricao.substring(1));
			if(valor.getDescricao().contains(descricao)) {
				return valor;
			}
		}
		
		return null;
	}
}

File: client/src/app/enumeration/tipo-contagem.ts
import { SelectItem } from "primeng/components/common/selectitem";

export enum TipoContagem {
    CONTAGEM_REFERENCIA = <any>"Contagem de Referência",
    ESTIMATIVA_INICIAL = <any>"Estimativa Inicial",
    CONTAGEM_ENCERRAMENTO = <any>"Contagem de Encerramento"
}

export namespace TipoContagem {
    export function getSelectItem(): any {
        var item: SelectItem[] = [];
        item.push({label:'Selecione', value:null});
        item.push({label:TipoContagem.CONTAGEM_REFERENCIA.toString(), value:TipoContagem[TipoContagem.CONTAGEM_REFERENCIA]});
        item.push({label:TipoContagem.ESTIMATIVA_INICIAL.toString(), value:TipoContagem[TipoContagem.ESTIMATIVA_INICIAL]});
        item.push({label:TipoContagem.CONTAGEM_ENCERRAMENTO.toString(), value:TipoContagem[TipoContagem.CONTAGEM_ENCERRAMENTO]});
        return item;
    } 
}